Lib/test/test_capi/test_opt.py

```diff
@@ -2660,7 +2662,38 @@ def f():
 
         f()
 
+    @requires_subprocess()
+    def test_interpreter_finalization_with_generator_alive(self):
+        code = textwrap.dedent("""
+            import sys
+            def simple_for():
+                for x in (1, 2):
+                    x
 
+            def gen():
+                try:
+                    yield
+                except:
+                    simple_for()
+
+            sys.settrace(lambda *args: None)
+            simple_for()
+            g = gen()
+            next(g)
+            print("finished")
+        """)
+
+        tmp = tempfile.NamedTemporaryFile(delete=False, suffix='.py')
+        tmp.write(code.encode('utf-8'))
+        tmp.close()
+        try:
+            p = subprocess.Popen([sys.executable, tmp.name], stdout=subprocess.PIPE, stderr=subprocess.STDOUT)
+            p.wait()
+            out = p.stdout.read()
+        finally:
+            os.remove(tmp.name)
+            p.stdout.close()
+        self.assertEqual(b"finished\n", out)
```

Comment:
Windows if failing due to CRLF differences. Please strip the newlines of the output and expected output.